### PR TITLE
security(SR-5): Add SSRF protection to portal

### PR DESCRIPTION
**Pull request by Zainzzkk**
## Introduction
Fix for SR-5. Portal need SSRF protection

## Resolution
* added function to check if mongoId is valid
* Added it to api file in portal which had ids which were constructed such as getDeal
* Added tests

### CODE REVIEW

**Comment by ChristianMcc** [comment]
There's an endpoint in `companies-house-api.js` that will also need protecting

File: portal/package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "start": "node server/index.js",
     "dev": "webpack --config webpack.config.dev.js --watch",
     "test": "jest --coverage --verbose --config=jest.config.js",
+    "test-dev": "jest --verbose --config=jest.config.js",
```
Comment (abhi-markan) [inline portal/package.json:19]: Is there a reason for this service?

File: portal/server/api-ssrf.test.js [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+require('dotenv').config();
+
+const api = require('./api');
+
+const { PORTAL_API_URL } = process.env;
+
+describe('api', () => {
+  const mock = new MockAdapter(axios);
+
+  const invalidId = '../../../etc/passwd';
```
Comment (abhi-markan) [inline portal/server/api-ssrf.test.js:12]: Can we also add following for invalid ID assertions:

* `127.0.0.1`
* []
* {}

File: portal/server/validation/validate-ids.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const validator = require('validator');
+
+/**
+ * isValidMongoId
+ * validates that a passed mongoId is a valid one
+ * returns true if so, false if not valid
+ * @param {String} mongoId
+ * @returns {Boolean}
```
Comment (abhi-markan) [inline portal/server/validation/validate-ids.js:8]: `@returns {Boolean} Ascertain whether MongoID is valid or not.`

File: portal/server/validation/validate-ids.test.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      expect(result).toEqual(false);
+    });
+
+    it('should return false if an id is not a valid mongo id', () => {
+      const result = isValidMongoId('12345');
+
+      expect(result).toEqual(false);
+    });
+
+    it('should return true if an id is a valid mongo id', () => {
```
Comment (abhi-markan) [inline portal/server/validation/validate-ids.test.js:17]: Please add following assertions:

* `127.0.0.1`
* `../../etc`
* {}
* []

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Few comments to address please.

File: portal/package.json (hunk 1 above)
Comment (Zainzzkk) [reply portal/package.json:19]: we have it in the others.  Useful when running tests locally as coverage takes up too much room

**Review by abhi-markan — DISMISSED** [review]

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**